### PR TITLE
Use image-loader import instead of download/upload in S3 watcher

### PR DESCRIPTION
**Pull request by theefer**
Rather than downloading and uploading the file via the Lambda, this uses the import endpoint of the loader to make the loader download the image directly from S3 using a time-limited URL.

As suggested by @kenoir.

### CODE REVIEW

File: s3watcher/lambda/lib/Logger.js [hunk 1]
```diff
@@ -35,15 +34,11 @@ module.exports = {
     messages: messages,
 
     log: function (stage, messageKey, state) {
-        log(stage, messageKey, state)
+        log(stage, messageKey, state);
```
Comment (akash1810) [inline s3watcher/lambda/lib/Logger.js]: :fearful: does this mean the linter didn't pick this up originally?!?!

File: s3watcher/lambda/lib/Transfer.js [hunk 2]
```diff
@@ -41,27 +41,27 @@ module.exports = {
...
-                    filename: upload.params.filename,
-                    uploadedBy: upload.params.uploadedBy,
-                    stage: upload.params.stage
-                });
+            Logger.logImport(config.stage, s3Object);
 
-                return Upload.postData(upload, data.Body);
-            }).retry(5).flatMap(function(uploadResult){
-                Logger.logRecordToCloudWatch(config.stage, uploadResult);
+            const urlExpiryInSeconds = config.s3UrlExpiry;
```
Comment (akash1810) [inline s3watcher/lambda/lib/Transfer.js]: The config object (and file in S3) is changing; is it worth thinking about how we do this?

File: s3watcher/lambda/lib/Upload.js [hunk 3]
```diff
@@ -20,10 +18,10 @@ module.exports = {
 
          const uploadedBy = buildUploadedBy(s3Event.path);
 
-         return  {
+         return {
              key: config.apiKey,
              url: config.baseUrl,
-             path: "/images",
+             path: "/imports",
              size: s3Event.size,
```
Comment (akash1810) [inline s3watcher/lambda/lib/Upload.js]: is `size` applicable now?

**Comment by akash1810** [comment]
couple of minor comments, but :+1:

File: s3watcher/lambda/lib/Logger.js (hunk 1 above)
Comment (theefer) [inline s3watcher/lambda/lib/Logger.js]: Does the linter run anywhere?

File: s3watcher/lambda/lib/Transfer.js (hunk 2 above)
Comment (theefer) [inline s3watcher/lambda/lib/Transfer.js]: Absolutely, I forgot to mention it.

File: s3watcher/lambda/lib/Upload.js (hunk 3 above)
Comment (theefer) [inline s3watcher/lambda/lib/Upload.js]: I thought it might get logged somewhere, but I'm not sure..?

Comment (akash1810) [inline s3watcher/lambda/lib/Upload.js]: I [does](https://github.com/guardian/grid/blob/master/s3watcher/lambda/lib/Transfer.js#L48), though this PR removes that.

I was thinking its more appropriate for image-loader to log this information now as it makes more sense in the context of that service.

Its minor though.

File: s3watcher/lambda/lib/Logger.js (hunk 1 above)
Comment (akash1810) [inline s3watcher/lambda/lib/Logger.js]: its not in the [ci script](https://github.com/guardian/grid/blob/master/s3watcher/build.sh)... oops! will raise a PR to fix.

File: s3watcher/lambda/lib/Transfer.js (hunk 2 above)
Comment (kenoir) [inline s3watcher/lambda/lib/Transfer.js]: Perhaps we should standardise on a way of doing this for Lambdas? I prefer the include config at build time option.

**Comment by kenoir** [comment]
Hopefully this will get rid of the errors we were seeing. 

Overall I prefer this approach as it makes the Lambda do even less work (pushing the possibility for error states out). I almost did this the first time round but decided against it as I didn't want to use the time limited URLs (in retrospect silly as the lambda is itself a time limited operation).

File: s3watcher/lambda/lib/Transfer.js (hunk 2 above)
Comment (theefer) [inline s3watcher/lambda/lib/Transfer.js]: Standardising would be nice, but I'm not sure about the build-time option. Shall we chat around coffee?

**Comment by theefer** [comment]
Come on, just admit it, you wanted to play with piping data through reactive streams ;-P

**Comment by kenoir** [comment]
@theefer obvs

**Pull request merged**